src/app/libp2p_helper/src/libp2p_helper/main.go

```diff
@@ -675,40 +676,52 @@ type incomingMsgUpcall struct {
 }
 
 func handleStreamReads(app *app, stream net.Stream, idx int) {
```

Comment:
I believe that reusing the same stream idx  here for multiple messages will require some OCaml side changes before we merge this PR.

Reply:
I see, I guess the current flow is 1 request -> 1 response -> close stream?

Reply:
At the moment, but I can update the OCaml code to handle this I think. This is a note for me to remember to do that :)